pulsar-broker/src/test/java/org/apache/pulsar/client/api/SimpleProducerConsumerTest.java

```diff
@@ -3692,4 +3693,88 @@ public void testGetStatsForPartitionedTopic() throws Exception {
         consumer.close();
         producer.close();
     }
+
+    @Test
+    public void testIncomingMessageSizeForNonPartitionedTopic() throws Exception {
+        final String topicName = "persistent://my-property/my-ns/testIncomingMessageSizeForNonPartitionedTopic-" +
+                UUID.randomUUID().toString();
+        final String subName = "my-sub";
+
+        @Cleanup
+        Consumer<byte[]> consumer = pulsarClient.newConsumer()
+                .topic(topicName)
+                .subscriptionName(subName)
+                .subscribe();
+
+        @Cleanup
+        Producer<byte[]> producer = pulsarClient.newProducer()
+                .topic(topicName)
+                .create();
+
+        final int messages = 100;
+        List<CompletableFuture<MessageId>> messageIds = new ArrayList<>(messages);
+        for (int i = 0; i < messages; i++) {
+            messageIds.add(producer.newMessage().value(("Message-" + i).getBytes()).sendAsync());
+        }
+        FutureUtil.waitForAll(messageIds).get();
```

Comment:
nitpick: in general, it's good to use timeouts in tests, for example using `.get(3, TimeUnit.SECONDS)`. However, I guess it's not very likely that sending would fail at this point, so the timeout isn't that relevant in this case.

Reply:
I don't think using timeout is a good idea. We should rely on the test framework timeout not the timeouts within the test.

Reply:
@sijie yes, I agree that placing timeouts all over could clutter the test code. 
Do you happen to know if there's already a default timeout set for each individual test method?

Reply:
Yes. The TestNG has global timeout settings.